src/main/java/com/github/fishio/PowerUpSpawner.java

```diff
@@ -0,0 +1,136 @@
+package com.github.fishio;
+
+import java.util.Random;
+
+import com.github.fishio.listeners.TickListener;
+import com.github.fishio.logging.Log;
+import com.github.fishio.logging.LogLevel;
+import com.github.fishio.power_ups.PowerUp;
+import com.github.fishio.power_ups.PuExtraLife;
+import com.github.fishio.power_ups.PuFreeze;
+import com.github.fishio.power_ups.PuSuperSpeed;
+
+/**
+ * A PowerUpSpawner repeatedly spawns random PowerUps
+ * in a PlayingField every certain interval.
+ */
+public class PowerUpSpawner implements TickListener {
+
+	private final PlayingField pf;
+	
+	private final int minX;
+	private final int maxX;
+	
+	public static final int WIDTH = 25;
+	public static final int HEIGHT = 25;
+	
+	private Log log;
+	
+	/**
+	 * The amount of different PowerUps that are supported.
+	 * This field should be updated with each new PowerUp added.
+	 */
+	private static final int POWERUP_COUNT = 3; 
+	
+	private Random rand = new Random();
+	
+	/**
+	 * The amount of seconds between each time
+	 * a PowerUp is spawned.
+	 */
+	private static final int INTERVAL = 30;
+
+	private final int intervalTicks;
+	private int tickCounter;
+	
+	/**
+	 * Creates a new PowerUpSpawner and automatically
+	 * registers it to the given PlayingField.
+	 * 
+	 * @param pf
+	 * 		The PlayingField this PowerUpSpawner
+	 * 		works on.
+	 */
+	public PowerUpSpawner(PlayingField pf) {
+		this.pf = pf;
+		pf.registerGameListener(this);
```

Comment:
Only expose "this" after constructing as much of yourself as possible. In this case, put registerGameListener on the last line of your constructor, so that all other fields are initialized before the playingField interacts with you.

<logic